### PR TITLE
ci: migrating to GitHub actions

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
Migrating to GitHub actions from CircleCI as per https://github.com/hypertrace/hypertrace/issues/144

### CODE REVIEW

**Review by aaron-steinfeld — COMMENTED** [review]
See comments from https://github.com/hypertrace/query-service/pull/47

**Comment by JBAhire** [comment]
Now using new custom actions for 
- Gradle
- Validate charts and publish

changes as per discussions here: https://github.com/hypertrace/query-service/pull/47

changes are tested here: https://github.com/hypertrace/attribute-service/pull/65 and merged here https://github.com/hypertrace/attribute-service/pull/66

File: .github/workflows/merge-publish.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+jobs:
+  merge-publish:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
```
Comment (kotharironak) [inline .github/workflows/merge-publish.yml:16]: This is needed only in `pull_request_target` context.

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: jacocoIntegrationTestReport
+
+      - name: copy test reports
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: copyAllReports --output-dir=/tmp/test-reports
+
+      - name: Upload coverage to Codecov
```
Comment (kotharironak) [inline .github/workflows/pr-test.yml:54]: We have to copy the integration test report too - right?
Need that step after integration test as well

Comment (JBAhire) [reply .github/workflows/pr-test.yml:54]: this copy runs after `integrationTest` is done and copies both unit test and integration test reports

File: .github/workflows/pr-test.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        with: 
+          args: jacocoTestReport
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          name: unit test reports
+          fail_ci_if_error: true
+          flags: unit
+
```
Comment (kotharironak) [inline .github/workflows/pr-test.yml:42]:
```suggestion
```

File: .github/workflows/pr-test.yml (hunk 2 above)
Comment (JBAhire) [reply .github/workflows/pr-test.yml:50]: I have moved it below codecov to follow standard ordering we are following everywhere

File: .github/workflows/pr-test.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Integration test
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: jacocoIntegrationTestReport
+  
+
```
Comment (kotharironak) [inline .github/workflows/pr-test.yml:48]:
```suggestion
```

**Review by kotharironak — APPROVED** [review]

**Pull request merged**